create_derivatives/routines.py

```diff
@@ -129,9 +133,136 @@ def ocr_pdf(self):
+                manifest_dir.mkdir()
+            self.fac.set_base_prezi_dir(str(manifest_dir))
+            self.create_manifest(jp2_files, manifest_dir, jp2_path, bag.dimes_identifier, bag.as_data)
+            bag.process_status = Bag.MANIFESTS_CREATED
+            bag.save()
+            bags_with_manifests.append(bag.dimes_identifier)
+        msg = "Manifests successfully created." if len(bags_with_manifests) else "No manifests created."
+        return msg, bags_with_manifests
+
+    def create_manifest(self, files, manifest_dir, image_dir, identifier,
+                        obj_data):
+        """Method that runs the other methods to build a manifest file and populate
+        it with information.
+
+        Args:
+            files (list): Files to iterate over
+            manifest_dir (str): Path to directory to write manifests to.
+            image_dir (str): Path to directory containing derivative image files.
+            identifier (str): A unique identifier.
+            obj_data (dict): Data about the archival object.
+        """
+        manifest_path = Path(manifest_dir, "{}.json".format(identifier))
+        page_number = 1
+        manifest = self.fac.manifest(ident="{}{}".format(self.fac.prezi_base, identifier), label=obj_data["title"])
+        manifest.set_metadata({"Date": obj_data["dates"]})
+        manifest.thumbnail = self.set_thumbnail(files[0].stem)
+        sequence = manifest.sequence(ident=identifier)
+        for file in files:
+            filename = file.stem
+            width, height = self.get_image_info(image_dir, file)
+            """Set the canvas ID, which starts the same as the manifest ID,
+            and then include page_number as the canvas ID.
+            """
+            canvas = sequence.canvas(
+                ident="{}/canvas/{}".format(
+                    manifest.id, str(page_number).zfill(3)),
+                label="Page {}".format(
+                    str(page_number)))
+            canvas.set_hw(height, width)
+            annotation = canvas.annotation(ident=filename)
+            img = annotation.image(
+                ident="/{}/full/max/0/default.jpg".format(filename))
+            self.set_image_data(img, height, width, filename)
+            canvas.thumbnail = self.set_thumbnail(filename)
+            page_number += 1
+        v2_json = manifest.toJSON(top=True)
+        v3_json = self.upgrader.process_resource(v2_json, top=True)
+        with open(manifest_path, 'w', encoding='utf-8') as jf:
+            json.dump(v3_json, jf, ensure_ascii=False, indent=4)
+
+    def get_image_info(self, image_dir, file):
+        """Gets information about the image file.
+
+        Args:
+            image_dir (str): path to the directory containing the image file
+            file (str): filename of the image file
+        Returns:
+            width (int): Pixel width of the image file
+            height (int): Pixel height of the image file
+        """
```

Comment:
Is this what the method is actually returning?

Also, this is more of a preference thing but you may want to call for the full filepath (or a Path object) instead of two arguments.